sdks/java/core/src/test/java/org/apache/beam/sdk/runners/inprocess/InMemoryWatermarkManagerTest.java

```diff
@@ -841,6 +852,46 @@ public void synchronizedProcessingInputTimeIsHeldToPendingBundleTimes() {
 
     clock.set(BoundedWindow.TIMESTAMP_MAX_VALUE);
     assertThat(downstreamWms.getSynchronizedProcessingInputTime(), equalTo(upstreamHold));
+
```

Comment:
Could/should this be in a separate test? Generally, more test methods is nice because we can provide better names/better signals about what is going on when some are green and some not.